### ISSUE TITLE
Set Password via link, wrong sender address

### ISSUE THREAD

**Issue by MSP-Stev** [owncloud/core#33440]
### Steps to reproduce
1. install a new owncloud 10.0.10 instance (mysql)
2. configure a mail server (Msexchange) with the following settings
a) send mode: smtp
b) encryption: none
c) from: cloud@domain1.com
d) auth method: Login - Authentication required
e) server: mail.domain1.com:25
f) credentials: domain1\cloud password
3. send test mail (after adding admin email address) -> all fine
4. add a new local user with email address newuser@domain2.com (external domain)
5. received "Your ownCloud account was created" mail from configured sender address (see 2c )
6. set password via the link in the mail
7. error pops up "Failed to send email. Please contact your administrator."

error log owncloud:

`{"reqId":"W@FgYoX@UF03k6xN26IMNwAAAAc","level":3,"time":"2018-11-05T16:49:34+00:00","remoteAddr":"****","user":"--","app":"settings","method":"POST","url":"\/index.php\/setpassword\/522030323417990203806\/test.user","message":"Exception: {\"Exception\":\"Swift_TransportException\",\"Message\":\"Expected response code 250 but got code \\\"550\\\", with message \\\"550 5.7.1 Client does not have permissions to send as this sender\\r\\n\\\"\",\"Code\":550,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/composer\\\/swiftmailer\\\/swiftmailer\\\/lib\\\/classes\\\/Swift\\\/Transport\\\/AbstractSmtpTransport.php(281): Swift_Transport_AbstractSmtpTransport->_assertResponseCode('550 5.7.1 Clien...', Array)\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/composer\\\/swiftmailer\\\/swiftmailer\\\/lib\\\/classes\\\/Swift\\\/Transport\\\/EsmtpTransport.php(270): Swift_Transport_AbstractSmtpTransport->executeCommand('\\\\r\\\\n.\\\\r\\\\n', Array, Array)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/composer\\\/swiftmailer\\\/swiftmailer\\\/lib\\\/classes\\\/Swift\\\/Transport\\\/AbstractSmtpTransport.php(333): Swift_Transport_EsmtpTransport->executeCommand('\\\\r\\\\n.\\\\r\\\\n', Array)\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/composer\\\/swiftmailer\\\/swiftmailer\\\/lib\\\/classes\\\/Swift\\\/Transport\\\/AbstractSmtpTransport.php(428): Swift_Transport_AbstractSmtpTransport->_streamMessage(Object(Swift_Message))\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/composer\\\/swiftmailer\\\/swiftmailer\\\/lib\\\/classes\\\/Swift\\\/Transport\\\/AbstractSmtpTransport.php(444): Swift_Transport_AbstractSmtpTransport->_doMailTransaction(Object(Swift_Message), 'test.user@do...', Array, Array)\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/composer\\\/swiftmailer\\\/swiftmailer\\\/lib\\\/classes\\\/Swift\\\/Transport\\\/AbstractSmtpTransport.php(176): Swift_Transport_AbstractSmtpTransport->_sendTo(Object(Swift_Message), 'test.user@do...', Array, Array)\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/Mail\\\/Mailer.php(99): Swift_Transport_AbstractSmtpTransport->send(Object(Swift_Message), Array)\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/settings\\\/Controller\\\/UsersController.php(717): OC\\\\Mail\\\\Mailer->send(Object(OC\\\\Mail\\\\Message))\\n#8 \\\/var\\\/www\\\/html\\\/owncloud\\\/settings\\\/Controller\\\/UsersController.php(685): OC\\\\Settings\\\\Controller\\\\UsersController->sendNotificationMail('test.user')\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(153): OC\\\\Settings\\\\Controller\\\\UsersController->setPassword('522030323417990...', 'test.user', 'odhjahsgfuasgd1...')\\n#10 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(85): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OC\\\\Settings\\\\Controller\\\\UsersController), 'setPassword')\\n#11 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(100): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OC\\\\Settings\\\\Controller\\\\UsersController), 'setPassword')\\n#12 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(46): OC\\\\AppFramework\\\\App::main('UsersController', 'setPassword', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#13 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/Route\\\/Router.php(342): OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#14 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/base.php(909): OC\\\\Route\\\\Router->match('\\\/setpassword\\\/52...')\\n#15 \\\/var\\\/www\\\/html\\\/owncloud\\\/index.php(54): OC::handleRequest()\\n#16 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/composer\\\/swiftmailer\\\/swiftmailer\\\/lib\\\/classes\\\/Swift\\\/Transport\\\/AbstractSmtpTransport.php\",\"Line\":383}"}`

-> ERROR: 5.7.1 Client does not have permissions to send as this sender

smtp log exchange:

```
2018-11-05T16:49:34.797Z,MSEXCHANGE,08D63B5629F3A7C3,56,IP:59638,*,SMTPSubmit SMTPAcceptAnyRecipient BypassAntiSpam AcceptRoutingHeaders,Set Session Permissions
2018-11-05T16:49:34.797Z,MSEXCHANGE,08D63B5629F3A7C3,57,IP:59638,>,250 XProxy accepted and authenticated,
2018-11-05T16:49:34.797Z,MSEXCHANGE,08D63B5629F3A7C3,58,IP:59638,<,MAIL FROM:<test.user@domain2.com>,
2018-11-05T16:49:34.797Z,MSEXCHANGE,08D63B5629F3A7C3,59,IP:59638,*,SMTPSubmit SMTPAcceptAnyRecipient BypassAntiSpam AcceptRoutingHeaders,Set Session Permissions
2018-11-05T16:49:34.797Z,MSEXCHANGE,08D63B5629F3A7C3,60,IP:59638,*,08D63B5629F3A7C3;2018-11-05T16:49:34.766Z;1,receiving message
2018-11-05T16:49:34.797Z,MSEXCHANGE,08D63B5629F3A7C3,61,IP:59638,>,250 2.1.0 Sender OK,
2018-11-05T16:49:34.797Z,MSEXCHANGE,08D63B5629F3A7C3,62,IP:59638,<,RCPT TO:<test.user@domain2.com>,
2018-11-05T16:49:34.797Z,MSEXCHANGE,08D63B5629F3A7C3,63,IP:59638,>,250 2.1.5 Recipient OK,
2018-11-05T16:49:34.797Z,MSEXCHANGE,08D63B5629F3A7C3,64,IP:59638,<,DATA,
2018-11-05T16:49:34.797Z,MSEXCHANGE,08D63B5629F3A7C3,65,IP:59638,>,354 Start mail input; end with <CRLF>.<CRLF>,
2018-11-05T16:49:34.891Z,MSEXCHANGE,08D63B5629F3A7C3,66,IP:59638,*,,receiving message with InternetMessageId <d798875846432c68e245ca06bc17e633@cloud.domain1.com>
2018-11-05T16:49:34.891Z,MSEXCHANGE,08D63B5629F3A7C3,67,IP:59638,>,550 5.7.1 Client does not have permissions to send as this sender,
2018-11-05T16:49:34.906Z,MSEXCHANGE,08D63B5629F3A7C3,68,IP:59638,<,QUIT,
```

### Expected behaviour
Set password should use the configured sender address.

### Actual behaviour
Set password uses the local user email address as sender address.

### Server configuration
**Operating system**:
CentOS 7 (3.10.0-862.14.4.el7.x86_64)

**Web server:**
Server version: Apache/2.4.6 (CentOS)
Server built:   Jun 27 2018 13:48:59

**Database:**
Server version: 5.5.60-MariaDB MariaDB Server

**PHP version:**
7.0.32

**ownCloud version:** (see ownCloud admin page)
version	10.0.10.4
versionstring	10.0.10
edition	Community
productname	ownCloud

**Updated from an older ownCloud or fresh install:**
New Installation (also same problem with updated installations)

**Where did you install ownCloud from:**
https://owncloud.org/download/

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks possibly related issues are https://github.com/owncloud/core/issues/927 (Invalid default (& lost password) email sender address), https://github.com/owncloud/core/issues/8826 ("Wrong Password" when successfully changing password), https://github.com/owncloud/core/issues/30511 (Password Email wont Set), https://github.com/owncloud/core/issues/19632 (Forgotten Share Link Password), and https://github.com/owncloud/core/pull/32002 (Report "Wrong current password" to user).

**Comment by ho4ho** [comment]
Looks like opposite of https://github.com/owncloud/core/issues/25470 ?

**Comment by phil-davis** [comment]
That open issue is a bit different. The problem here is that the email should come from the "proper" email address of the server. And it people really notice the error if their email server refuses to accept/process and email pretending to be from some "external" address.

This has been fixed recently, I remember an acceptance test being added. Someone could find the older issue and link here?

**Comment by paurakhsharma** [comment]
I believe this was fixed here https://github.com/owncloud/user_management/pull/91. Backport here: #33167

**Comment by PVince81** [comment]
fix will be in 10.1.0, you might try again with the 10.1.0 beta
